pkg/proxy/iptables/proxier.go

```diff
@@ -266,7 +266,7 @@ func NewProxier(ipFamily v1.IPFamily,
 	masqueradeMark := fmt.Sprintf("%#08x", masqueradeValue)
 	klog.V(2).InfoS("Using iptables mark for masquerade", "ipFamily", ipt.Protocol(), "mark", masqueradeMark)
 
-	serviceHealthServer := healthcheck.NewServiceHealthServer(hostname, recorder, nodePortAddresses, healthzServer)
+	serviceHealthServer := healthcheck.NewServiceHealthServer(hostname, recorder, nodePortAddresses, healthManager)
```

Comment:
So this means we are still creating separate IPv4 and IPv6 ServiceHealthServers which track the separate IPv4 and IPv6 ProxierHealthManagers, meaning we still have problems. I know you said "this will _partly_ fix #116486" but I wanted to make sure you were aware of this...

Reply:
I saw two dual-stack kube-proxy issues https://github.com/kubernetes/kubernetes/issues/116486 (ProxierHealthServer) and https://github.com/kubernetes/kubernetes/issues/114702 (ServiceHealthServers) and I planned to have different PRs for both the issues. 

Let me look into the ServiceHealthServers and see if I can work on that in this PR only.

Reply:
ah, right, I forgot there were two issues. I don't think this PR makes the situation with ServiceHealthServer any _worse_, so it's OK if we don't fix it yet